### ISSUE TITLE
Ships engineer not assigned properly when Hire Full Compliment used on Space Stations

### ISSUE THREAD

**Issue by pheonixstorm** [MegaMek/mekhq#3]
0.3.23

After buying a space station, using hire full compliment will not assign a tech to some space stations. After several tests it seems that this only happens with stations that do not have gunners. Normal crew assignments work fine.

This issue will go away if you un-assign one crew member and then reassign them to the station.
[Station test.zip](https://github.com/MegaMek/mekhq/files/98617/Station.test.zip)

**Comment by pheonixstorm** [comment]
Small update. The unassign/assign does not last. When you save/reload the campaign the assigned tech is blank again.

**Comment by pheonixstorm** [comment]
Update 2.
This bug also seems to cause repairs to the affected station to be shorthanded.

**Comment by Dylan-M** [comment]
Space stations are not fully supported at this time.

**Comment by pheonixstorm** [comment]
Yeah, just checked with 0.3.18 as I thought it was related to the commit I posted about in the first ticket but happens prior to those changes as well.

**Comment by pheonixstorm** [comment]
Question. Is there any reason a space station does not have a minimum of one pilot? They do have station keeping drives so someone competent should be at the helm.

**Comment by mkerensky** [comment]
Still an issue as of 0.43.4.   It would seem that the unarmed station doesn't count as a unit or its personnel don't count as people -  Unit.resetEngineer never triggers for the unarmed station, though it does for all other units - mechs, other spacecraft, etc. I think it actually triggers on a per-person basis.  Digging continues...

**Comment by mkerensky** [comment]
Fixed in my current PR dealing with edge for non-combat rolls. Had to do some other work on the ship's engineer anyway, so it made sense to add this in too.  Unit.calcCompositeCrew() was returning empty because the unarmed stations lacked both pilots and gunners, thus triggering getCrew().isMissing() in Entity.  I added a check to ignore space stations for this line, but adding at least 1 pilot to the station crew would be another possible fix, and I agree with your point that space stations should have pilots if JumpShips do.  I would be inclined to add 3 just like all other large craft have.
